luigi/task.py

```diff
@@ -324,7 +337,19 @@ def __hash__(self):
         return self.__hash
 
     def __repr__(self):
-        return self.task_id
+        params = self.get_params()
+        param_values = self.get_param_values(params, [], self.param_kwargs)
+
+        # Build up task id
+        task_id_parts = []
+        param_objs = dict(params)
+        for param_name, param_value in param_values:
+            if param_objs[param_name].significant:
```

Comment:
I'm undecided.  Maybe people will want to keep repr() short by marking configuration parameters as insignificant?  I don't use this feature so I don't really have an opinion.